src/Language/PureScript/Pretty/Types.hs

```diff
@@ -32,7 +32,7 @@ typeLiterals = mkPattern match
   where
   match TypeWildcard{} = Just $ text "_"
   match (TypeVar var) = Just $ text var
-  match (TypeLevelString s) = Just . text $ show s
+  match (TypeLevelString s) = Just $ text s
```

Comment:
Unfortunately this will cause problems when type level string is rendered outside of a `Fail`.